src/settings/__tests__/settingsReducer-test.js

```diff
@@ -12,6 +12,8 @@ import { EventTypes } from '../../api/eventTypes';
 import settingsReducer from '../settingsReducer';
 import * as eg from '../../__tests__/lib/exampleData';
 
+/* eslint jest/expect-expect: ["error", { "assertFunctionNames": ["expect", "check"] }] */
```

Comment:
This doesn't have an effect -- the rule is only checking that a `test` callback contains an `expect` call. Here, the `check` calls come outside `test`, not inside, and the `test` callbacks already contain `expect`. So we can just leave this line out.

The rule we'd want which would have caught this issue would be that a `describe` callback contains a `test` call. I don't see such a rule in the plugin, unfortunately:
https://github.com/jest-community/eslint-plugin-jest#rules

There is [no-standalone-expect](https://github.com/jest-community/eslint-plugin-jest/blob/7c28c6b9588506b9d11f9c5d148c96e97831e86a/docs/rules/no-standalone-expect.md), sort of a converse, which would also have caught this… but in our eslintrc we disabled it because it's buggy:
```yaml
  # The docs for this rule [1] say that placing an `expect` call
  # outside a `test` or `it` block (e.g., in a `beforeEach` or
  # `afterEach`) means that the assertion isn't executed. Empirically,
  # this seems just wrong [2], and there are a few places where we
  # want to call `expect` in a `beforeEach` or `afterEach` to assert
  # that multiple tests in the same file aren't interfering with each
  # other by leaving bits of state lying around.
  #
  # [1] https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-standalone-expect.md
  # [2] https://github.com/zulip/zulip-mobile/pull/4235#discussion_r489984717
  jest/no-standalone-expect: off
```
(Also from its docs it sounds like the rule wouldn't actually have caught this anyway? "It is viable, however, to have an `expect` in a helper function that is called from within a `test` or `it` block so `expect` statements in a function will not trigger this rule.")

Reply:
Ah, right, thanks for the investigation!